### ISSUE TITLE
Style/UnlessLogicalOperators reporting false offenses

### ISSUE THREAD

**Issue by Incanus3** [rubocop/rubocop#9551]
After upgrading rubocop to 1.11.0 I started getting

```ruby
C: Style/UnlessLogicalOperators: Do not use mixed logical operators in an unless.
```

on a line that reads

```ruby
unless values[:client][:insurance_company] == 999
```

I don't think there's a logical operator in this condition (equality operator is a boolean operator, but not a logical one AFAIK) and they definitely aren't mixed.

--------

## Expected behavior

No offense reported.

## Actual behavior

When running `rubocop --debug` I get the following output (filtered out lines starting with "Scanning" and "Loading cache"):
```
For /home/jakub/Projects/spital/covid-form-backend: configuration from /home/jakub/Projects/spital/covid-form-backend/.rubocop.yml
configuration from /home/jakub/.rbenv/versions/2.7.2/gemsets/covid/gems/rubocop-performance-1.10.0/config/default.yml
configuration from /home/jakub/.rbenv/versions/2.7.2/gemsets/covid/gems/rubocop-performance-1.10.0/config/default.yml
Default configuration from /home/jakub/.rbenv/versions/2.7.2/gemsets/covid/gems/rubocop-1.11.0/config/default.yml
configuration from /home/jakub/.rbenv/versions/2.7.2/gemsets/covid/gems/rubocop-rake-0.5.1/config/default.yml
configuration from /home/jakub/.rbenv/versions/2.7.2/gemsets/covid/gems/rubocop-rake-0.5.1/config/default.yml
configuration from /home/jakub/.rbenv/versions/2.7.2/gemsets/covid/gems/rubocop-rspec-2.2.0/config/default.yml
configuration from /home/jakub/.rbenv/versions/2.7.2/gemsets/covid/gems/rubocop-rspec-2.2.0/config/default.yml
Inspecting 89 files
For /home/jakub/Projects/spital/covid-form-backend/app/persistence/migrations: configuration from /home/jakub/Projects/spital/covid-form-backend/app/persistence/migrations/.rubocop.yml
Inheriting configuration from /home/jakub/Projects/spital/covid-form-backend/.rubocop.yml
AllCops/Exclude configuration from /home/jakub/Projects/spital/covid-form-backend/.rubocop.yml
For /home/jakub/Projects/spital/covid-form-backend/spec: configuration from /home/jakub/Projects/spital/covid-form-backend/spec/.rubocop.yml
Inheriting configuration from /home/jakub/Projects/spital/covid-form-backend/.rubocop.yml
AllCops/Exclude configuration from /home/jakub/Projects/spital/covid-form-backend/.rubocop.yml
For /home/jakub/Projects/spital/covid-form-backend/spec/helpers: configuration from /home/jakub/Projects/spital/covid-form-backend/spec/helpers/.rubocop.yml
Inheriting configuration from /home/jakub/Projects/spital/covid-form-backend/spec/.rubocop.yml
Inheriting configuration from /home/jakub/Projects/spital/covid-form-backend/.rubocop.yml
AllCops/Exclude configuration from /home/jakub/Projects/spital/covid-form-backend/.rubocop.yml
.

Offenses:

app/web/validation/contracts.rb:64:13: C: Style/UnlessLogicalOperators: Do not use mixed logical operators in an unless.
            unless values[:client][:insurance_company] == 999 ...
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

89 files inspected, 1 offense detected
Finished in 0.4577370000188239 seconds
```

## Steps to reproduce the problem

Create a file with the following contents:

```ruby
unless values[:client][:insurance_company] == 999
  key.failure(Messages.must_not_be_shorter(than: 9))

  if (match = value.match(INSURANCE_NUMBER_REGEX))
    month = match[:month]

    if month < 1 || month > 12
      key.failure(Messages.not_a_valid_month(month))
    else
      puts 'whatever'
    end
  else
    puts 'whatever'
  end
end
```

and run `rubocop` on it with the rule enabled. The code is actually pretty nonsensical as I tried to get a minimal failing example while also avoiding triggering other offenses. The original code is [here](https://github.com/Incanus3/covid-form-backend/blob/394356f3064bbb2a224f66623721938cd81584f5/app/web/validation/contracts.rb#L64). The interesting thing here is that if I remove any of the nested blocks, rubocop stops reporting this offense. On the other hand, if I just commend something out, the offense persists.

## RuboCop version

```
❯ bundle exec rubocop -V
1.11.0 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 2.7.2 x86_64-linux)
  - rubocop-performance 1.10.0
  - rubocop-rake 0.5.1
  - rubocop-rspec 2.2.0
```

**Comment by Incanus3** [comment]
can confirm that this fixes the issue. thanks for the hard work guys.

**Comment by bbatsov** [comment]
You're welcome!
